fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/rule/AggregateFilterScanRule.java

```diff
@@ -0,0 +1,26 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.rule.transformation.materialization.rule;
+
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+/*
+ *
+ * Here is the rule for pattern Aggregate-Filter-Scan
+ *
+ */
+public class AggregateFilterScanRule extends AggregateRewriteBaseRule {
```

Comment:
Why filter not merge into scan

Reply:
maybe the subtree is from mv rewrite. so there is pattern aggregate-filter

Reply:
In mv, the compensation predicates are added to a Filter operator.